content/bash-scripting-basics.md

```diff
@@ -0,0 +1,158 @@
+
+The second crucial detail for your OS to properly run your script is to make it executable. As of right now, your script is just a regular text file. In order for your OS to run your script, we will need to make this text file `executable`. This can be done by navigating to our `~/scripts/` directory with our File Manager. Once you locate your script file, right click on the file to bring up with context menu and select "Properties". This will bring up the various details about the file we are looking at. Click on the tab labelled "Permissions". There will be a checkbox labelled "Allow executing file as program". Make sure to check this box. Once this is enabled, the script will be able to run.
+
+At the moment, the script will do nothing if we decided to execute it, so let's add some output and then test our script. Open the script and add the following line below your first `shebang` line:
+
+```bash
+echo "The script executed correctly."
+```
+
+We can now attempt to run our script. Open the Terminal from your Application Launcher. I am going to assume that you have followed along and created a directory in your Home Directory called `scripts` and that you have named this script `basics.sh` (if you have placed it in a separate directory or named it differently, the steps are the same, but you must substitute the commands accordingly). First, we need to navigate in our Terminal to our `scripts` directory. Input the following command in your Terminal to change directory to `~/scripts`:
+
+```bash
+cd ~/scripts
+```
+
+You can now run your script by running the following command in your Terminal:
+
+```bash
+./basics.sh
+```
+
+If it worked, your terminal should output the following line:
+
+`The script executed correctly.`
+
+Congratulations! You have created a working BASH script. Of course, outputting a single line does not provide much functionality, so we will be doing just that in the next section.
+
+## Adding functionality
+
+At this point, our script does one thing: It outputs a single line in our Terminal. This works, but it does not provide any use case. There are many things you can do with a script if you can get creative and combine several commands together to create something useful. There is a lot to cover, and there are too many things that can be created for me to cover in this. If you would like to explore more ideas, a good place to find useful commands you can use in your script is by searching for what you need to accomplish, and then after finding the appropriate command, add it to a script.
```

Comment:
> searching for what you need to accomplish, and then after finding the appropriate command, add it to a script.

Searching where? A web search?

Also, this might be a good place to mention that they can run the same commands in a terminal if they just want to test things out quickly without having to save a file and call a script repeatedly.

Reply:
This was not addressed